src/js/player.js

```diff
@@ -2881,31 +2881,38 @@ class Player extends Component {
 
     this.setInterval(function() {
       // Check to see if mouse/touch activity has happened
-      if (this.userActivity_) {
-        // Reset the activity tracker
-        this.userActivity_ = false;
-
-        // If the user state was inactive, set the state to active
-        this.userActive(true);
-
-        // Clear any existing inactivity timeout to start the timer over
-        this.clearTimeout(inactivityTimeout);
-
-        const timeout = this.options_.inactivityTimeout;
-
-        if (timeout > 0) {
-          // In <timeout> milliseconds, if no more activity has occurred the
-          // user will be considered inactive
-          inactivityTimeout = this.setTimeout(function() {
-            // Protect against the case where the inactivityTimeout can trigger just
-            // before the next user activity is picked up by the activity check loop
-            // causing a flicker
-            if (!this.userActivity_) {
-              this.userActive(false);
-            }
-          }, timeout);
-        }
+      if (!this.userActivity_) {
+        return;
+      }
+
+      // Reset the activity tracker
+      this.userActivity_ = false;
+
+      // If the user state was inactive, set the state to active
+      this.userActive(true);
+
+      // Clear any existing inactivity timeout to start the timer over
+      this.clearTimeout(inactivityTimeout);
+
+      const timeout = this.options_.inactivityTimeout;
+
+      // Min time for setTimeout is 4ms regardless any less number
+      // https://www.w3.org/TR/2011/WD-html5-20110525/timers.html#timers
+      if (timeout <= 4) {
```

Comment:
I think we want to keep the `0` check here because while the timer does resolve to 4ms, the value `0` has the special meaning of stopping the `inactivityTimeout`. A number that isn't zero means that you want a timeout. I think it would be confusing if you set the `inactivityTimeout` to 2 and then have it not function.

Reply:
Hi, does that mean ```timeout === 0``` good enough or it should be at normal ```timeout < 0```, I'm in flavor the first if just 0 if treat it as special

Reply:
unfortunately, we should keep the same check because it could break people. We can correct this in a major update.